### PR TITLE
add function for climate forecasts

### PR DESCRIPTION
**Pull request by sdtaylor**
monthly mean temperature and total precip forecasts with a 6 month lead time, obtained from https://climate.northwestknowledge.net/RangelandForecast/download.php

#158, #249

### CODE REVIEW

File: tools/forecast_tools.R [hunk 1]
```diff
@@ -379,3 +379,53 @@ forecast_viz <- function(obs_data, obs_date_col_name, obs_val_col_name, for_data
     geom_line(data = for_data_sub, mapping = aes_string(x = for_date_col_name, y = for_val_col_name), color = "blue") +
     labs(x='',y=ylabel)
 }
+
+####################################################################################
+#' Download downscaled climate forecast data for a single location
+#' 
+#' Obtained from https://climate.northwestknowledge.net/RangelandForecast/download.php
+#' 
+#' @param climate_model Individual climate models available are are c('CFSv2','CMC1','CMC2','GFDL-FLOR','GFDL','NASA','NCAR'),
```
Comment (ethanwhite) [inline tools/forecast_tools.R]: Remove extra `are` and move model list down a line to keep lines shorter.

File: tools/forecast_tools.R [hunk 2]
```diff
@@ -379,3 +379,53 @@ forecast_viz <- function(obs_data, obs_date_col_name, obs_val_col_name, for_data
...
+#' @param lon longitude Default is for Portal, AZ
+#' 
+#' @return a data.frame with precipitation(mm), temperature(C), year, and month. Temperature is the mean temperature
+#'         for the month, while precipitation is the total forecasted precip.
+
+get_climate_forecasts = function(climate_model = 'ENSMEAN', 
+                                 lat = 31.9555, lon = -109.0744,
+                                 lead_time = 6){
+  
+  if(!climate_model %in% c('CFSv2','CMC1','CMC2','GFDL-FLOR','GFDL','NASA','NCAR','ENSMEAN')) stop(paste0('Unknown climate model: ',climate_model))
```
Comment (ethanwhite) [inline tools/forecast_tools.R]: Let's keep the lines at a more readable length here.

```R
models = c('CFSv2', 'CMC1', 'CMC2', 'GFDL-FLOR', 'GFDL', 'NASA', 'NCAR', 'ENSMEAN')
if(!climate_model %in% models){
  stop(paste0('Unknown climate model: ',climate_model))
}
```

File: tools/forecast_tools.R [hunk 3]
```diff
@@ -379,3 +379,53 @@ forecast_viz <- function(obs_data, obs_date_col_name, obs_val_col_name, for_data
...
+  if(!lead_time %in% 1:7) stop(paste0('Lead time must an integer be between 1 and 7, got: ',lead_time))
```
Comment (ethanwhite) [inline tools/forecast_tools.R]: Move to use of `{}` here as well for length.

File: tools/forecast_tools.R [hunk 4]
```diff
@@ -379,3 +379,53 @@ forecast_viz <- function(obs_data, obs_date_col_name, obs_val_col_name, for_data
...
+  today = Sys.Date()
+  start_time = strftime(today, format='%Y-%m-%d')
+  end_time = strftime(today %m+% months(lead_time), format='%Y-%m-%d')
+  
+  # add in timestamps for URL
+  start_time = paste0(start_time,'T00%3A00%3A00Z')
+  end_time = paste0(end_time,'T00%3A00%3A00Z')
+  
+  full_url = paste0('https://tds-proxy.nkn.uidaho.edu/thredds/ncss/NWCSC_INTEGRATED_SCENARIOS_ALL_CLIMATE/bcsd-nmme/monthlyForecasts/bcsd_nmme_metdata_',climate_model,
+                    '_forecast_1monthAverage.nc?var=prate&var=tmp2m&latitude=',lat,'&longitude=',lon,'&time_start=',start_time,'&time_end=',end_time,'&accept=csv')
```
Comment (ethanwhite) [inline tools/forecast_tools.R]: I think this reads a little easier:

```R
base_url = 'https://tds-proxy.nkn.uidaho.edu/thredds/ncss/NWCSC_INTEGRATED_SCENARIOS_ALL_CLIMATE/bcsd-nmme/monthlyForecasts/bcsd_nmme_metdata_'
full_url = paste0(base_url, climate_model,
		  '_forecast_1monthAverage.nc?var=prate&var=tmp2m',
		  '&latitude=', lat, '&longitude=', lon,
		  '&time_start=', start_time, '&time_end=',end_time,
		  '&accept=csv')
```

**Review by ethanwhite — COMMENTED** [review]
LGTM. A few suggestions to keep the line length down in a couple of spots and make things easier to follow.

**Comment by sdtaylor** [comment]
Ok, I put in all your suggestions @ethanwhite

**Review by ethanwhite — APPROVED** [review]
Thanks @sdtaylor! LGTM.

@gmyenni why don't you take a look and merge when you're happy.

**Comment by gmyenni** [comment]
Sweeet!!

**Pull request merged**